app/src/main/java/uk/gov/onelogin/navigation/NavigatorImpl.kt

```diff
@@ -0,0 +1,42 @@
+package uk.gov.onelogin.navigation
+
+import android.util.Log
+import androidx.navigation.NavHostController
+import javax.inject.Inject
+import uk.gov.onelogin.developer.DeveloperRoutes.navigateToDeveloperPanel
+
+class NavigatorImpl @Inject constructor() : Navigator {
+    private var navController: NavHostController? = null
+        get() {
+            if (field == null) Log.w(this::class.simpleName, "Navigator not initialised")
```

Comment:
What happens if the navController is not initialised here? Would there be any situation where this would be possible?

Reply:
There shouldn't be, it should be linked to the lifecycle of the Activity. If `navigate()` is called and it is not set then it will at least make a warning log, it won't crash the app, it simply just won't navigate.

Reply:
I would argue that it would be better to have a crash, as I think of it as:
- A problem a developer needs to be aware of -> crash
- A problem a user needs to be aware of -> Recoverable path with Error Screen 
but I don't actually have a problem with how you have done this. I think it is fine